src/io/exporters/FBXInfoExtractor.js

```diff
@@ -0,0 +1,324 @@
+          color2,
+        },
+        index,
+      },
+      matrix,
+    } = mesh;
+
+    const model = new FBXModel();
+    const instCount = mesh.geometry.maxInstancedCount;
+    const oneCCylinder = new fbxgeo.OneColorGeo();
+    oneCCylinder.init(mesh.geometry);
+    const splittingInfo = this._gatherCylindersColoringInfo(mesh.geometry);
+    let twoCCylinder = null;
+    if (splittingInfo.needToSplit > 0) {
+      twoCCylinder = new fbxgeo.TwoColoredCylinder();
+      twoCCylinder.init(mesh.geometry, splittingInfo);
+    }
+    const additionalVertsCount = splittingInfo.addPerCylinder * splittingInfo.needToSplit;
+    const vertCount = position.count;
+    const indsCount = index.count;
+    model.init(instCount * vertCount + additionalVertsCount, instCount * indsCount);
+    const instMatrix = new THREE.Matrix4();
+    const objMatrix = new THREE.Matrix4();
+    const colorStart = new THREE.Color();
+    const colorEnd = new THREE.Color();
+    let geo = {};
+    for (let instanceIndex = 0; instanceIndex < instCount; ++instanceIndex) {
+      // update colors in geometry
+      const colorIdx = instanceIndex * color.itemSize;
+      if (splittingInfo.is2Colored[instanceIndex]) {
+        // .color2 contains starting color, and .color contains starting color (see uber.frag ATTR_COLOR2)
+        colorStart.fromArray(color2.array, colorIdx);
+        colorEnd.fromArray(color.array, colorIdx);
+        twoCCylinder.setColors(colorStart, colorEnd);
+        geo = twoCCylinder;
+      } else {
+        // has one color per cylinder
+        colorStart.fromArray(color.array, colorIdx);
+        oneCCylinder.setColors(colorStart);
+        geo = oneCCylinder;
+      }
+      // add instance to the model
+      this._getCylinderInstanceMatrix(mesh.geometry, instanceIndex, instMatrix);
+      objMatrix.multiplyMatrices(matrix, instMatrix);
+      model.addInstance(objMatrix, geo);
+    }
+    const material = this._collectMaterialInfo(mesh);
+    this._addToPool(model, material);
+  }
+
+  /**
+   * Adding model to pool of models or extend existing ones
+   * @param {object} model - model to add
+   * @param {object} material - material to add
+   */
+  _addToPool(model, material) {
+    const materialIdx = this._checkExistingMaterial(material);
+    if (materialIdx === this._models.length) { // new model-material pair
+      model.reworkIndices();
+      this._models.push(model);
+      this._materials.push(material);
+    } else { // add model to existing model-material pair
+      const oldModel = this._models[materialIdx];
+      oldModel.concatenate(model);
+    }
+  }
+
+  /**
+   * Checking if given material already was registered in materials pool (no need to create new one)
+   * @param {object} material - given material
+   * @returns {number} number of model-material pair
+   */
+  _checkExistingMaterial(material) {
+    for (let i = 0; i < this._materials.length; ++i) {
+      if (_.isEqual(material, this._materials[i])) {
```

Comment:
Should we consider using [`_.findIndex()`](https://lodash.com/docs/#findIndex) or similar for clarity?

Reply:
Yes, it's better. Done.